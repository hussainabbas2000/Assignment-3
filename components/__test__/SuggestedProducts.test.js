/**
 * @jest-environment jsdom
 */
 import React from "react";
 import ReactDOM  from "react-dom";
 import SuggestedProducts from "../product/SuggestedProducts";
 import {render,cleanup} from '@testing-library/react';
 import "@testing-library/jest-dom/extend-expect"
 import renderer from "react-test-renderer";
 afterEach(cleanup);
 it("renders Suggested Products without crash",()=>{
     const div = document.createElement("div");
     ReactDOM.render(<SuggestedProducts product = "1"></SuggestedProducts>,div); 
 })
 it("renders Suggested Products correctly",()=>{
     const {getByTestId} = render(<SuggestedProducts product = "1"/>)
     const element = getByTestId("suggested")
     expect(element).not.toHaveClass()
 
 })
 
 it("matches snapshot 2",()=>{
     const item = renderer.create(<SuggestedProducts product = "1"></SuggestedProducts>).toJSON();
     expect(item).toMatchSnapshot();
 })