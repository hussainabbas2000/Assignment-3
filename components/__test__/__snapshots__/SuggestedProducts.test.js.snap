// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<div
  data-testid="suggested"
>
  <h3
    className="MuiTypography-root MuiTypography-h6"
  >
    Suggested Products
  </h3>
  <div
    className="RSFSuggestedProducts-products-9"
  >
    <div
      className="RSFLoadMask-root-12 RSFLoadMask-show-16"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={{}}
          />
        </svg>
      </div>
    </div>
    <div
      className="RSFSuggestedProducts-wrap-10"
    />
  </div>
</div>
`;
